src/rust/zh.manhuaren/src/lib.rs

```diff
@@ -0,0 +1,460 @@
+			cover,
+			title: manga_obj
+				.get("mangaName")
+				.as_string()
+				.unwrap_or_else(|_| StringRef::from(""))
+				.read(),
+			author: manga_obj
+				.get("mangaAuthor")
+				.as_string()
+				.unwrap_or_else(|_| StringRef::from(""))
+				.read(),
+			artist: manga_obj
+				.get("mangaAuthor")
+				.as_string()
+				.unwrap_or_else(|_| StringRef::from(""))
+				.read(),
+			description: String::new(),
+			url: String::new(),
+			categories: Vec::new(),
+			status: match manga_obj.get("mangaIsOver").as_int().unwrap_or(-1) {
+				0 => MangaStatus::Ongoing,
+				1 => MangaStatus::Completed,
+				_ => MangaStatus::Unknown,
+			},
+			nsfw: MangaContentRating::Safe,
+			viewer: MangaViewer::Vertical,
+		});
+	}
+
+	Ok(MangaPageResult {
+		manga: manga_arr,
+		has_more: item_count > 0,
+	})
+}
+
+fn get_manga_list_by_query(query: String, page: i32) -> Result<MangaPageResult> {
+	let mut args: Vec<(String, String)> = vec![
+		(String::from("keywords"), query),
+		(
+			String::from("start"),
+			helper::i32_to_string((page - 1) * 20),
+		),
+		(String::from("limit"), String::from("20")),
+	];
+
+	let qs = helper::generate_get_query(&mut args);
+
+	let url = String::from(API_URL) + "/v1/search/getSearchManga?" + &qs;
+
+	let body = helper::request(url, HttpMethod::Get)?;
+
+	let json = json::parse(body)?.as_object()?;
+	let response = json.get("response").as_object()?;
+	let mangas = response.get("result").as_array()?;
+	let item_count = mangas.len();
+
+	let mut manga_arr: Vec<Manga> = Vec::new();
+
+	for manga in mangas {
+		let manga_obj = manga.as_object()?;
+
+		manga_arr.push(Manga {
+			id: helper::i32_to_string(manga_obj.get("mangaId").as_int().unwrap_or(0) as i32),
+			cover:
+				// this cover val won't be empty
+				manga_obj
+				.get("mangaCoverimageUrl")
+				.as_string()
+				.unwrap_or_else(|_| StringRef::from(""))
+				.read(),
+			title: manga_obj
+				.get("mangaName")
+				.as_string()
+				.unwrap_or_else(|_| StringRef::from(""))
+				.read(),
+			author: manga_obj
+				.get("mangaAuthor")
+				.as_string()
+				.unwrap_or_else(|_| StringRef::from(""))
+				.read(),
+			artist: manga_obj
+				.get("mangaAuthor")
+				.as_string()
+				.unwrap_or_else(|_| StringRef::from(""))
+				.read(),
+			description: String::new(),
+			url: String::new(),
+			categories: Vec::new(),
+			status: match manga_obj.get("mangaIsOver").as_int().unwrap_or(-1) {
+				0 => MangaStatus::Ongoing,
+				1 => MangaStatus::Completed,
+				_ => MangaStatus::Unknown,
+			},
+			nsfw: MangaContentRating::Safe,
+			viewer: MangaViewer::Vertical,
+		});
+	}
+
+	Ok(MangaPageResult {
+		manga: manga_arr,
+		has_more: item_count > 0,
+	})
+}
+
+fn parse_chapters(manga: &ObjectRef, key: &str) -> Vec<Chapter> {
+	match manga.get(key).as_array() {
+		Ok(chapters) => {
+			let mut chapter_arr: Vec<Chapter> = Vec::new();
+
+			for ch in chapters {
+				let ch_obj = ch.as_object().unwrap();
```

Comment:
```suggestion
				let ch_obj = ch.as_object().expect("chapter item should be object");
```
or equivalent error message